internal/receiver/databricksreceiver/internal/databricks/metrics_provider_test.go

```diff
@@ -0,0 +1,147 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package databricks
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/signalfx/splunk-otel-collector/internal/receiver/databricksreceiver/internal/commontest"
+)
+
+func TestDatabricksMetricsProvider(t *testing.T) {
+	var dbrsvc = NewService(&testdataRawClient{testDataDir: commontest.TestdataDir}, 25)
+	mp := MetricsProvider{Svc: dbrsvc}
+
+	builder := commontest.NewTestMetricsBuilder()
+	_, err := mp.AddJobStatusMetrics(builder, 0)
+	require.NoError(t, err)
+	emitted := builder.Emit()
+	assert.Equal(t, 3, emitted.MetricCount())
+
+	metricMap := commontest.MetricsByName(emitted)
+	rms := emitted.ResourceMetrics()
+	assert.Equal(t, 1, rms.Len())
+	rm := rms.At(0)
+	sms := rm.ScopeMetrics()
+
+	assert.Equal(t, 1, sms.Len())
+
+	const dbrjt = "databricks.jobs.total"
+	jobTotalMetrics := metricMap[dbrjt]
+	assert.Equal(t, dbrjt, jobTotalMetrics.Name())
+	assert.EqualValues(t, 6, jobTotalMetrics.Gauge().DataPoints().At(0).IntValue())
+
+	const dbrjss = "databricks.jobs.schedule.status"
+	jobScheduleMetrics := metricMap[dbrjss]
+	assert.Equal(t, dbrjss, jobScheduleMetrics.Name())
+	pts := jobScheduleMetrics.Gauge().DataPoints()
+	assert.Equal(t, 6, pts.Len())
+	assert.EqualValues(t, 0, pts.At(0).IntValue())
+
+	const dbrtss = "databricks.tasks.schedule.status"
+	taskStatusMetric := metricMap[dbrtss]
+	assert.Equal(t, dbrtss, taskStatusMetric.Name())
+	taskPts := taskStatusMetric.Gauge().DataPoints()
+	assert.Equal(t, 8, taskPts.Len())
+
+	taskMap := tasksToMap(taskPts)
```

Comment:
not asking atm but standardizing on https://pkg.go.dev/github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatatest/pmetrictest#CompareMetrics could be helpful for maintainability and comprehension in the future.

Reply:
Interesting. Noted, thanks.